urlff74b96b0d_HugoFeng_HospitalProject_tgz-pJ8-hugo_project_hospital_ProjectTestJ8/src/hugo/util/structure/Vector.java

```diff
@@ -0,0 +1,188 @@
+/**
+ * @author Xuyang Feng
+ * @email hugo.fxy@gmail.com
+ */
+
+package hugo.util.structure;
+import java.util.Iterator;
+
+@SuppressWarnings("rawtypes")
+public class Vector<E extends Comparable> implements Iterable<E>
+{
+	protected Object data[];
+	protected int count;
+	
+	public Vector(int capacity)
+	{
+		data = new Object[capacity];
+		count = 0;
+	}
+
+	public int size()
+	{
+		return count;
+	}
+ 
+	public boolean isEmpty()
+	{
+		return size() == 0;
+	}
+
+	public Object get(int index)
+	{
+		return data[index];
+	}
+
+	public void set(int index, Object obj)
+	{
+		data[index] = obj;
+	}
+
+	public boolean contains(Object obj)
+	{
+		for(int i=0;i<count;i++)
+		{
+			if(data[i] == obj) return true;
+		}
+		return false;
+	}
+	
+	public void addFirst(Object item)
+	{
+		if (data.length==size()) {
+			this.extendCapacity();
+		}
+		for (int i = count; i>0; i--) {
+			data[count] = data[count-1];
+		}
+		data[0] = item;
+		count++;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 52:59, 64:71.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.